### PR TITLE
2.3 Image enhancements 

### PR DESCRIPTION
**Pull request by djdefi**
- [x] Run more things as www-data user
- [x] Use `ENV` in Dockerfile to specify Cachet version (simplify new release process)
- [x] Run php-fpm workers on-demand rather than preforking
- [x] disable supervisor http socket (not used)
- [x] Take advantage of [ARG](https://docs.docker.com/engine/reference/builder/#arg) in Dockerfile to allow specifying `cachet_ver` during manual builds

Merging these fixes back into master, then will cut a 2.3.0 tag and release when it is released upstream.

### CODE REVIEW

File: docker/php-fpm-pool.conf [hunk 1]
```diff
@@ -6,11 +6,10 @@ listen = 9000
 
 request_terminate_timeout = 120s
 
-pm = dynamic
+pm = ondemand
```
Comment (GrahamCampbell) [inline docker/php-fpm-pool.conf]: what at the advantages here?

Comment (djdefi) [inline docker/php-fpm-pool.conf]: Minor reduction in RAM utilization on an idle system. Performance should be about the same overall.

Per https://ma.ttias.be/a-better-way-to-run-php-fpm/

Comment (GrahamCampbell) [inline docker/php-fpm-pool.conf]: hmmm ,ok

**Comment by djdefi** [comment]
Ran through tests of the app, everything seems to be working just fine. There was a issue with a 500 when saving an incident on the pre 2.3enhance image iteration which is not happening now.

**Pull request merged**